pkg/engine/turn/modify.go

```diff
@@ -1,48 +1,64 @@
 	"github.com/simimpact/srsim/pkg/key"
 )
 
+// SetGauge sets the gauge of the target as detailed in the input ModifyAttribute struct.
+// 1. Update gauge of target
+// 2. Move target to top of order
+// 3. Sort target down based on AV. In the event of tie, this target should be at top of order.
+//		If there is an active turn and it is not this target, this target should go below the active
+//		turn (so index 1 instead of 0 when 0 gauge/AV)
+// 4. Emit GaugeChangeEvent
 func (mgr *manager) SetGauge(data info.ModifyAttribute) error {
-	prev := mgr.target(data.Target).gauge
+
+	previousGauge := mgr.target(data.Target).gauge
 	mgr.target(data.Target).gauge = data.Amount
 
-	if mgr.target(data.Target).gauge == prev {
+	// if there's no change to Gauge, exit early
+	if mgr.target(data.Target).gauge == previousGauge {
 		return nil
 	}
 
-	// create map of TargetID -> AV so you only need to calc once within this call
-	targetAV := make(map[key.TargetID]float64, len(mgr.order.order))
-
-	// TODO:
-	// 1. update gauge of target
-	// 2. move target to top of order
-	// 3. sort target down based on AV. In the event of tie, this target should be at top of order.
-	//		If there is an active turn and it is not this target, this target should go below the active
-	//		turn (so index 1 instead of 0 when 0 gauge/AV)
-	// 4. emit GaugeChangeEvent
+	// find target index in mgr.orderHandler.turnOrder
+	targetIndex := 0
+	for i, v := range mgr.orderHandler.turnOrder {
+		if data.Target == v.id {
+			targetIndex = i
+			break
+		}
+	}
 
-	// TODO: this is also needed for TurnStart emit & TurnReset emit, should be abstracted
-	status := make([]event.TurnStatus, len(mgr.order.order))
-	for i, t := range mgr.order.order {
-		status[i] = event.TurnStatus{
-			ID:    t.id,
-			Gauge: t.gauge,
-			AV:    targetAV[t.id],
-			// TODO: should we also add speed to this?
+	// targetIndex == 0 indicates its already at the start of turnOrder, so no change needs to be made.
+	// if there is an activeTurn, set our target to index 1; otherwise set to index 0.
+	if targetIndex == 0 {
+	} else if !mgr.activeTurn {
+		if targetIndex + 1 < mgr.orderHandler.Len() {
+			mgr.orderHandler.turnOrder = append([]*target{mgr.target(data.Target)}, append(mgr.orderHandler.turnOrder[:targetIndex], mgr.orderHandler.turnOrder[targetIndex+1:]...)...)
+		} else {
+			mgr.orderHandler.turnOrder = append([]*target{mgr.target(data.Target)}, mgr.orderHandler.turnOrder[:targetIndex]...)
+		}
+	} else {
+		if targetIndex + 1 < mgr.orderHandler.Len() {
+			mgr.orderHandler.turnOrder = append([]*target{mgr.orderHandler.turnOrder[0]}, append([]*target{mgr.target(data.Target)}, append(mgr.orderHandler.turnOrder[1:targetIndex], mgr.orderHandler.turnOrder[targetIndex+1:]...)...)...)
+		} else {
+			mgr.orderHandler.turnOrder = append([]*target{mgr.orderHandler.turnOrder[0]}, append([]*target{mgr.target(data.Target)}, mgr.orderHandler.turnOrder[1:targetIndex]...)...)
 		}
 	}
```

Comment:
My suggestion is change this to the following steps:
1. Set `turnOrder` to be a new array with the target removed (`append(mgr.orderHandler.turnOrder[:targetIndex], mgr.orderHandler.turnOrder[targetIndex+1:]...)`)
2. Add the target to the top of the array
3. flip index 0 and index 1 iff `mgr.activeTarget != data.Target`

Alternatively, can go with a solution that only needs to loop over the array once and shuffles everything down in-place. Would look something like this (not tested, may not be correct):
```go
startIndex := 0
if mgr.activeTurn && targetIndex != 0 {
  startIndex = 1
}

prev := mgr.orderHandler.turnOrder[targetIndex] // use mgr.target return if saved from another call
for i := startIndex; i <= targetIndex; i++ {
  tmp := mgr.orderHandler.turnOrder[i]
  mgr.orderHandler.turnOrder[i] = prev
  prev = tmp
}
```